src/Microsoft.DotNet.Build.Tasks/PackageFiles/CodeCoverage.targets

```diff
@@ -131,6 +91,38 @@
   </Target>
   <!-- *********************************************************************************************** -->
 
+  <Target Name="CreateCoverageDedicatedRuntime"
+    BeforeTargets="GenerateTestExecutionScripts"
+    DependsOnTargets="GenerateWindowsPdbsForAssemblyBeingTested"
+    Condition="'$(CodeCoverageEnabled)'=='true' and '$(UseCoverageDedicatedRuntime)'=='true'">
+    <PropertyGroup>
+      <CoverageDedicatedRuntimeDir>$(TestHostRootPath)shared/Microsoft.NETCore.App/10.10.10</CoverageDedicatedRuntimeDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <RuntimeFiles Include="$(NETCoreAppTestSharedFrameworkPath)/*.*" />
+      <RuntimeFiles Include="$(NETCoreAppTestSharedFrameworkPath)il/*.*" />
+      <RuntimeFilesAmd64 Include="$(NETCoreAppTestSharedFrameworkPath)amd64/*.*" />
+      <RuntimeFilesx86 Include="$(NETCoreAppTestSharedFrameworkPath)x86/*.*" />
+      <RuntimeWindowsPdbs Include="$(NETCoreAppTestSharedFrameworkPath)WindowsPDB/*.*" />
+    </ItemGroup>
+    <Copy SourceFiles="@(RuntimeFiles)"
+          DestinationFolder="$(CoverageDedicatedRuntimeDir)"
+          SkipUnchangedFiles="true"
+          UseHardlinksIfPossible="false" />
+    <Copy SourceFiles="@(RuntimeFilesAmd64)"
```

Comment:
Is there any expected overwriting happening? If so it is probably worth considering producing the filelist in an item group to merge them and remove duplicates before doing the copy to avoid overwriting.

Reply:
I expect that to happen just to System.Private.CoreLib all others should not cause any overwrite.